### PR TITLE
[Security] Handshaker status return

### PR DESCRIPTION
**Pull request by gtcooke94**
Address the first (easy) half of https://github.com/grpc/grpc/issues/32598

### CODE REVIEW

**Review by matthewstevenson88 — APPROVED** [review]
LGTM, but please wait for @markdroth's approval before merging.

File: src/core/lib/security/transport/security_handshaker.cc
Comment (markdroth) [inline src/core/lib/security/transport/security_handshaker.cc:663]: Please change `handshaker.status()` to:

```
absl::Status(
    handshaker.status().code(),
    absl::StrCat("Failed to create security handshaker: ", handshaker.status().message())
```

This way, we're adding a useful prefix to the status message but leaving the code as-is.

File: src/core/lib/security/transport/security_handshaker.cc
Comment (markdroth) [inline src/core/lib/security/transport/security_handshaker.cc:666]: Let's stick with the existing message "Failed to create security handshaker:".

File: src/core/lib/security/transport/security_handshaker.cc
Comment (markdroth) [inline src/core/lib/security/transport/security_handshaker.cc:665]: Please add a TODO that once all TSI implementations are updated to pass a `StatusOr<>` instead of null, we should change this to use `absl::InternalError()`.

File: src/core/lib/security/transport/security_handshaker.cc
Comment (markdroth) [inline src/core/lib/security/transport/security_handshaker.cc:595]: Once you make the changes I suggested below, we won't need to do this here.  We can remove the local `error` variable and just directly use `status_` in its place.

File: src/core/lib/security/transport/security_handshaker.cc
Comment (markdroth) [inline src/core/lib/security/transport/security_handshaker.cc:667]: Please move this TODO to line 670.

File: src/core/lib/security/transport/security_handshaker.cc
Comment (markdroth) [reply src/core/lib/security/transport/security_handshaker.cc:595]: We don't need to call `GRPC_ERROR_CREATE()` at all.  Just pass `status_` to `ExecCtx::Run()` instead of `error`.

Note that `grpc_error_handle` used to be a custom type, but these days it's actually a typedef for `absl::Status`.  But there's a lot of legacy machinery for adding structured annotations to the `absl::Status` to simulate the legacy behavior of `grpc_error_handle`, and we're trying to slowly get rid of that tech debt in the code base.  `GRPC_ERROR_CREATE()` is one of the functions that add those legacy annotations, which is why I'm saying that we should get rid of it here.

File: src/core/lib/security/transport/security_handshaker.cc
Comment (gtcooke94) [reply src/core/lib/security/transport/security_handshaker.cc:667]: Done

File: src/core/lib/security/transport/security_handshaker.cc
Comment (gtcooke94) [reply src/core/lib/security/transport/security_handshaker.cc]: > these days it's actually a typedef for absl::Status

Ah this was my missing link, changed

File: src/core/lib/security/transport/security_handshaker.cc
Comment (markdroth) [inline src/core/lib/security/transport/security_handshaker.cc:606]: Data members need to come after the dtor.

https://google.github.io/styleguide/cppguide.html#Declaration_Order

**Review by markdroth — APPROVED** [review]
Just one remaining nit, we can get this landed once that's addressed.

File: src/core/lib/security/transport/security_handshaker.cc
Comment (gtcooke94) [reply src/core/lib/security/transport/security_handshaker.cc:606]: Fixed

**Review by markdroth — APPROVED** [review]

**Pull request closed without merging**
